File: src/main/java/kurenai/mybot/config/BotProperties.java
package kurenai.mybot.config;

import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;

/**
 * @author liufuhong
 * @since 2021-06-30 14:10
 */

@Data
@ConfigurationProperties(prefix = "bot")
public class BotProperties {
  private String token;
  private String username;
}

File: src/main/kotlin/kurenai/imsyncbot/config/BotProperties.kt
package kurenai.imsyncbot.config

import org.springframework.boot.context.properties.ConfigurationProperties
import java.util.*

@ConfigurationProperties(prefix = "bot")
class BotProperties {

    var ban = Ban()

    class Ban {
        var group: List<Long> = Collections.emptyList()
        var member: List<Long> = Collections.emptyList()
    }

}